tidy3d/components/tcad/data/monitor_data/charge.py

```diff
@@ -0,0 +1,220 @@
+"""Monitor level data, store the DataArrays associated with a single heat-charge monitor."""
+
+from __future__ import annotations
+
+from typing import Optional
+
+import pydantic.v1 as pd
+
+from tidy3d.components.base import skip_if_fields_missing
+from tidy3d.components.data.data_array import ElectroStaticDataArray
+from tidy3d.components.tcad.data.monitor_data.abstract import (
+    FieldDatasetTypes,
+    HeatChargeDataset,
+    HeatChargeMonitorData,
+)
+from tidy3d.components.tcad.monitors.charge import (
+    StaticCapacitanceMonitor,
+    StaticChargeCarrierMonitor,
+    StaticVoltageMonitor,
+)
+from tidy3d.constants import VOLT
+from tidy3d.log import log
+
+# Isn't this just static voltage?
+# class PotentialData(HeatChargeMonitorData):
+#     """Class that stores electric potential from a charge simulation."""
+#
+#     monitor: StaticVoltageMonitor = pd.Field(
+#         ...,
+#         title="Voltage monitor",
+#         description="Electric potential monitor associated with a Charge simulation.",
+#     )
+#
+#     voltage_series: HeatChargeDataset = pd.Field(
+#         None, title="Voltage series", description="Contains the voltages."
+#     )
+#
+#     @pd.validator("voltage_series", always=True)
+#     @skip_if_fields_missing(["monitor"])
+#     def warn_no_data(cls, val, values):
+#         """Warn if no data provided."""
+#
+#         mnt = values.get("monitor")
+#
+#         if val is None:
+#             log.warning(
+#                 f"No data is available for monitor '{mnt.name}'. This is typically caused by "
+#                 "monitor not intersecting any solid medium."
+#             )
+#
+#         return val
+#
+#     @property
+#     def symmetry_expanded_copy(self) -> PotentialData:
+#         """Return copy of self with symmetry applied."""
+#
+#         new_voltages = self._symmetry_expanded_copy(property=self.voltage_series.field_series)
+#
+#         return self.updated_copy(
+#             voltage_series=self.voltage_series.updated_copy(field_series=new_voltages)
+#         )
+#
+#     def field_name(self, val: str) -> str:
+#         """Gets the name of the fields to be plot."""
+#         if val == "abs^2":
+#             return "|V|²"
+#         else:
+#             return "V"
+
+
+class StaticChargeCarrierData(HeatChargeMonitorData):
+    """Class that stores free carrier concentration in Charge simulations."""
```

Comment:
Maybe `DCFreeCarrierData`?

Reply:
One reason why the static naming maybe makes sense is due to the temporal nature of our simulations, which when they are multi-physical then this changes. For example, free carriers may change on a static heat simulation, independent of a DC voltage/current. As such static represents this temporal relationship clearly I think.